services/statestorage/service.go

```diff
@@ -108,5 +115,9 @@ func (s *service) GetStateStorageBlockHeight(input *services.GetStateStorageBloc
 }
 
 func (s *service) GetStateHash(input *services.GetStateHashInput) (*services.GetStateHashOutput, error) {
-	panic("Not implemented")
+	value, _ := s.merkle.GetRoot(merkle.RootId(input.BlockHeight))
```

Comment:
Can't this fail? What happens if the block height is out of the forrest? like I give a block height of 1 million when the system has 10 blocks

Reply:
In the current implementation it will return an empty byte array.

Now after BlockTracker is in master we can add the latch here and return an error in case we are out of range.